src/main/java/com/dropbox/core/android/AuthActivity.java

```diff
@@ -529,10 +617,35 @@ protected void onNewIntent(Intent intent) {
             }
 
             // Successful auth.
-            newResult = new Intent();
-            newResult.putExtra(EXTRA_ACCESS_TOKEN, token);
-            newResult.putExtra(EXTRA_ACCESS_SECRET, secret);
-            newResult.putExtra(EXTRA_UID, uid);
+            if (token.equals(TokenType.OAUTH2.toString())) {
+                // token flow
+                newResult = new Intent();
+                newResult.putExtra(EXTRA_ACCESS_TOKEN, token);
+                newResult.putExtra(EXTRA_ACCESS_SECRET, secret);
+                newResult.putExtra(EXTRA_UID, uid);
+            } else if (token.equals(TokenType.OAUTH2CODE.toString())) {
+                // code flow with PKCE
+                TokenRequest tokenRequest = new TokenRequest(secret);
+                try {
+                    DbxAuthFinish dbxAuthFinish = tokenRequest.execute().get();
+
+                    if (dbxAuthFinish == null) {
+                        newResult = null;
+                    } else {
+                        newResult = new Intent();
+                        newResult.putExtra(EXTRA_ACCESS_TOKEN, dbxAuthFinish.getAccessToken());
+                        newResult.putExtra(EXTRA_ACCESS_SECRET, dbxAuthFinish.getAccessToken());
```

Comment:
The token and secret are the same. Is this intended? If so we should probably add a comment here explaining why they are the same.

Reply:
Agree. Actually access token and access secret it the OAuth1 concept. In this SDK EXTRA_ACCESS_TOKEN is not really used. I will remove them.